pyccel/ast/datatypes.py

```diff
@@ -81,7 +81,20 @@
     "logical" : {
         4  : "C_BOOL"}
 }
-
+iso_c_binding_shortcut_mapping = {
+    'C_INT8_T'              : 'i8',
+    'C_INT16_T'             : 'i16',
+    'C_INT32_T'             : 'i32',
+    'C_INT64_T'             : 'i64',
+    'C_INT128_T'            : 'i128',
+    'C_FLOAT'               : 'f32',
+    'C_DOUBLE'              : 'f64',
+    'C_LONG_DOUBLE'         : 'f128',
+    'C_FLOAT_COMPLEX'       : 'C_FLOAT_COMPLEX',
+    'C_DOUBLE_COMPLEX'      : 'C_DOUBLE_COMPLEX',
+    'C_LONG_DOUBLE_COMPLEX' : 'C_LONG_DOUBLE_COMPLEX',
```

Comment:
can't we have shorter names for those types too?

Reply:
We don't need shorter names for these types. The shorter names are there to make the code less verbose when writing literals (i.e. `3_C_INT`) however the only way to write a complex literal in is to write 2 real literals, so the complex datatypes are only used in declaration and type casting. Here it is not a problem if we are explicit rather than succinct

Reply:
ah, now I see.

Reply:
What about variables' declarations? Don't we want to use the shortcuts there too?

```f90
function add_complex(a, b) result(c)
    complex(c128), value :: a  ! shortcut
    complex(c128), value :: b  ! shortcut
    complex(c128) :: c         ! shortcut
    c = a + b
end function
```

Reply:
If you think that this is better then we can. It will then at least all match. We didn't do this already because we were not certain of what the standard abbreviation was for complexes (different options were discussed on slack [here](https://pyccel.slack.com/archives/C018QDHENUU/p1629120620053900)). I didn't think it was a big problem if we were a little more verbose in the declarations

Reply:
So, shall i always abbreviate complex datatypes? shall i use c32, c64, and c128?

Reply:
I don't think it is important what abbreviation we use for complex numbers, as we can always change our dictionary in the future. But if our goal is to make the code less verbose, than I suggest that we do it systematically also for the variable declarations.

Reply:
> So, shall i always abbreviate complex datatypes? shall i use c32, c64, and c128?

Yes, I think that `c64`, `c128`, `c256`  and so forth should be clear enough and will not generate any confusion for the user.